tests/4.5/task/test_task_Mergeable.c

```diff
@@ -0,0 +1,92 @@
+//===-- test_task_Mergeable.c -------------------------------------------===//
+//
+// OpenMP API Version 4.5 Nov 2015
+//
+// Description
+// Test case description: This test case demonstrates that mergeable clause
+// combined with if clause
+//===----------------------------------------------------------------------===//
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include "omp.h"
+#include "ompvv.h"
+#define THREADS 64
+#define MERGE 0  //  0 is for merge & 1 is for No merge
+
+
+struct Node {
+  int NodPos;
+  int MasterThrd;
+  int ExcutnThrd;
+  struct Node *Nxt;
+};
+
+
+int main() {
+  int errors = 0;
+  omp_set_num_threads(THREADS);
+
+  // Creating Linked list
+  struct Node *Head = NULL;
+  for (int i = 0; i < 10; ++i) {
+    struct Node *temp = malloc(sizeof(struct Node));
+    temp->Nxt = NULL;
+    if (Head == NULL) {
+      Head = temp;
+    } else {
+      struct Node *tmp = Head;
+      while (tmp != NULL) {
+        if (tmp->Nxt == NULL) {
+          tmp->Nxt = temp;
+          break;
+        }
+        tmp = tmp->Nxt;
+      }
+    }
+  }
+
+
+#pragma omp parallel
+  {
+    struct Node *Tmp = Head;
+    int Count = 1;
+#pragma omp single
+    {
+      int OutThrdId = omp_get_thread_num();
+      while (Tmp != NULL) {
+        Count++;
+#pragma omp task mergeable if (MERGE) firstprivate(Count)
```

Comment:
OpenMP V4.5 says `When the mergeable clause is present on a task construct, the generated task is a mergeable
task.` and the definition of the mergeable task is `A task that may be a merged task if it is an undeferred task or an included task.` Does this mean that a task with a mergeable clause always becomes a merged task? Or it can be a merged task?
This question is about the meaning of the word `may` in the mergeable task definition. In OpenACC, the word `may` indicates permission, but I'm not sure whether OpenMP has the same meaning or not.
If OpenMP also uses the word `may` as permission, the error-checking in line 90 can be wrong.

Reply:
This test case is passing on AMD platforms. Please let me know if this test is failing on your end. Due to inherent ambiguity of this comment, and the fact that it passes on AMD platform, I consider this comment as resolved for now...

Reply:
@lthakur007 , can you verify the number of threads executing the parallel region ?
There are few problems with this test:
1) The one that @seyonglee pointed -- "may" means eligible NOT that they have to merge
2) The mergable tasks are the ones created inside of the while and may be executed by any thread of the parallel team (NOT limited to the thread creating the tasks)
Example: Say num_threads are 4, Thread 1 creates the 10 tasks and thread 2 merges them and executes them. Your current code will error out.